src/pymoca/tree.py

```diff
@@ -298,7 +315,8 @@ def find_name(
     """
     # Look for ast.Class or ast.Symbol per the MLS v3.5:
     # 1. Simple Name Lookup (spec 5.3.1)
-    #     0. Predefined types (`Real`, `Integer`, `Boolean`, `String`) (spec 4.8)
+    #     0.1 Predefined types (`Real`, `Integer`, `Boolean`, `String`) (spec 4.8)
```

Comment:
I don't really see this mentioned in the spec, except the seemingly "opposite" statement is there

> In the latter case the lookup stops except for the predefined types, functions and operators defined in this specification. For these cases the lookup continues in the global scope, where they are defined.

That is (of course 😅) not what OpenModelica does:

```Modelica
model A
    model Boolean
        constant Integer bla = 0;
    end Boolean;

    Boolean b;
end A;
```

I can't seem to find a way to refer to _my_ version of `Boolean`, OpenModelica just takes the built-in one.

Reply:
The spec that you quoted is for the specific case of when lookup stops due to `encapsulated`.

See section 4.8 (per the comment on line 320 :smile:):

> The names Real, Integer, Boolean and String are reserved such that it is illegal to declare an element with these names.

There are other builtin types (e.g. `Clock`) that *can* be hidden by user definitions.